src/serde_utils.rs

```diff
@@ -0,0 +1,50 @@
+/// Make "string-like" implementations of `serde::Deserialize` and
+/// `serde::Serialize` for the given type, using that type's `Display` and
+/// `FromStr` impls.
+///
+/// (This is used for example to serialize codons as `"ACG"` rather than
+/// `["A","C","G"]`. We could equivalently use `serde_with::{DeserializeFromStr,
+/// SerializeDisplay}`, but it's not worth the extra dependency.)
+macro_rules! impl_stringlike {
+    ($type:ty) => {
+        #[cfg(feature = "serde")]
+        impl<'de> serde::Deserialize<'de> for $type {
+            fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+            where
+                D: serde::Deserializer<'de>,
+            {
+                struct Helper;
+                impl<'de> serde::de::Visitor<'de> for Helper {
+                    type Value = $type;
+
+                    fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+                        write!(formatter, "a string")
+                    }
+
+                    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+                    where
+                        E: serde::de::Error,
+                    {
+                        value
+                            .parse::<Self::Value>()

```

Comment:
Feel free to disregard this suggestion as I recognize it's a matter of taste, but IMHO the small size of the function makes it easy to infer from the return type that it must be parsing as a `Self::Value`.

```suggestion
                            .parse()
```